src/formatDate.js

```diff
@@ -8,7 +8,68 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  if (typeof date !== 'string' || date.length === 0) {
+    return '';
+  }
+
+  if (
+    !fromFormat ||
+    !toFormat ||
+    typeof fromFormat !== 'object' ||
+    typeof toFormat !== 'object' ||
+    !('length' in fromFormat) ||
+    !('length' in toFormat)
+  ) {
+    return '';
```

Comment:
Since the task specifies that input validation is not needed, the checks for the `date`, `fromFormat`, and `toFormat` arguments are unnecessary and can be removed. This will simplify the function and keep it focused on the core functionality.